packages/cli-kit/src/public/node/plugins/tunnel.ts

```diff
@@ -33,15 +47,38 @@ export interface HookReturnPerTunnelPlugin {
 
 export type TunnelProviderFunction = FanoutHookFunction<'tunnel_provider', ''>
 export type TunnelStartFunction = FanoutHookFunction<'tunnel_start', ''>
+export type TunnelStopFunction = FanoutHookFunction<'tunnel_start', ''>
+export type TunnelStatusFunction = FanoutHookFunction<'tunnel_status', ''>
+
 export type TunnelStartReturn = PluginReturnsForHook<'tunnel_start', ''>
+export type TunnelStopReturn = PluginReturnsForHook<'tunnel_start', ''>
+export type TunnelStatusReturn = PluginReturnsForHook<'tunnel_status', ''>
+
 export type TunnelStartAction = (port: number) => Promise<TunnelStartReturn>
+export type TunnelStatusAction = () => Promise<TunnelStatusReturn>
+export type TunnelStopAction = () => TunnelStopReturn
```

Comment:
I'd be more flexible here and return `Promise<TunnelStopReturn>` here like we are doing with status and start.